### ISSUE TITLE
Debian repository signing key for `packages.element.io` expires on 2023-04-15

### ISSUE THREAD

**Issue by richvdh** [vector-im/element-web#23786]
The debian installation instructions for element desktop, at https://element.io/get-started#linux-details, suggest downloading the key at https://packages.element.io/debian/element-io-archive-keyring.gpg and configuring apt to trust it.

This key will expire on 2023-04-15. Before then it will be necessary to update the key and get everyone to download it again.

**Comment by richvdh** [comment]
One way to get everyone to update would be to build and distribute a `keyring` package containing the key and the apt config.

**Comment by richvdh** [comment]
See also https://github.com/matrix-org/synapse/issues/10389

**Comment by davidegirardi** [comment]
> One way to get everyone to update would be to build and distribute a `keyring` package containing the key and the apt config.

So a keyring package signed by the current key, containing the old and new key at first, then just the current key until it is time to renew again?

**Comment by richvdh** [comment]
Typically we don't *replace* the key, just reissue the same on with an extended validity period. So: a keyring package signed by the current key, containing the current key.

**Comment by davidegirardi** [comment]
But having a package would help us in case we need to rotate a key too, right?

**Comment by TNTBOMBOM** [comment]
Im getting this error at the moment when i try to update element repo: 

```
E: The repository 'https://packages.element.io/debian default InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```
I have deleted repo and old key and then re-add them again follow latest website instructions but still the same result. 

So wonder if something changed resulted into this error.

**Comment by t3chguy** [comment]
@TNTBOMBOM that is unrelated to this issue, I suggest you open a new issue and fill all the required fields.

**Comment by richvdh** [comment]
Just for clarity, since it confused me for a while:

> The debian installation instructions for element desktop, at https://element.io/get-started#linux-details, suggest downloading the key at https://packages.element.io/debian/element-io-archive-keyring.gpg

This is *different* to the key at https://packages.element.io/element-release-key.gpg.

**Comment by t3chguy** [comment]
Thanks for your help with this @richvdh - new keyring has been deployed, along with a keyring package included in 1.11.25 onwards, I will ensure some announcement/docs are available for anything updating from <1.11.25 after the expiration date of the old keyring

**Comment by t3chguy** [comment]
If when running `apt update` you see a message like the following

```
Err:4 https://packages.element.io/debian default InRelease
  The following signatures were invalid: EXPKEYSIG C2850B265AC085BD riot.im packages <packages@riot.im>
Reading package lists... Done
W: GPG error: https://packages.element.io/debian default InRelease: The following signatures were invalid: EXPKEYSIG C2850B265AC085BD riot.im packages <packages@riot.im>
E: The repository 'https://packages.element.io/debian default InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

![image](https://user-images.githubusercontent.com/2403652/231256873-e0e31240-7064-44ff-b8c8-80ab13ba8b05.png)


Run `sudo wget -O /usr/share/keyrings/element-io-archive-keyring.gpg https://packages.element.io/debian/element-io-archive-keyring.gpg` and `apt update` should work once again

**Comment by philclifford** [comment]
@t3chguy Sadly not working for me : 
```
W: Failed to fetch https://packages.riot.im/debian/dists/default/InRelease  The following signatures were invalid: EXPKEYSIG C2850B265AC085BD riot.im packages <packages@riot.im>
W: Some index files failed to download. They have been ignored, or old ones used instead.
gpg -k --no-default-keyring --keyring /usr/share/keyrings/element-io-archive-keyring.gpg
/usr/share/keyrings/element-io-archive-keyring.gpg
--------------------------------------------------
pub   rsa4096 2019-04-15 [SC] [expires: 2033-03-13]
      12D4CD600C2240A9F4A82071D7B0B66941D01538
uid           [ unknown] riot.im packages <packages@riot.im>
sub   rsa3072 2019-04-15 [S] [expires: 2025-03-15]
```

**Comment by stayen** [comment]
@t3chguy Alas, that didn't compute:

    W: GPG error: https://packages.element.io/debian default Release: The following signatures were invalid: EXPKEYSIG C2850B265AC085BD riot.im packages <packages@riot.im>
    E: The repository 'https://packages.riot.im/debian default Release' is no longer signed.
    N: Updating from such a repository can't be done securely, and is therefore disabled by default.

**Comment by t3chguy** [comment]
@philclifford & @stayen what does your `/etc/apt/sources.list.d/element-io.list` say? Likely the path to the keyring it is pointing to is different than the instructions at https://element.io/download#linux

**Comment by eighthave** [comment]
This worked for me:
```console
$ grep signed-by /etc/apt/sources.list.d/packages_riot_im_debian.list 
deb [arch=amd64 signed-by=/usr/share/keyrings/element-desktop-keyring.gpg] https://packages.riot.im/debian/ bullseye main
$ sudo gpg --no-default-keyring \
	     --keyring /usr/share/keyrings/element-desktop-keyring.gpg \
	     --refresh-keys
```

**Comment by philclifford** [comment]
Thanks @eighthave : the `--refresh-keys` was key !
@t3chguy I have the same new keyring in `/usr/share/keyrings/element-io-archive-keyring.gpg`and `/usr/share/keyrings/element-desktop-archive-keyring.gpg` and my current sources.list.d entry points to the first of those (but I tried reverting to the instructions with no joy either) . Only after the `--refresh-keys` apt is finally happy again.

**Comment by stayen** [comment]
@t3chguy 

```
# cat /etc/apt/sources.list.d/element-io.list
deb [signed-by=/usr/share/keyrings/element-io-archive-keyring.gpg] https://packages.element.io/debian/ default main

```

On https://element.io/download#linux

`echo "deb [signed-by=/usr/share/keyrings/element-io-archive-keyring.gpg] https://packages.element.io/debian/ default main" | sudo tee /etc/apt/sources.list.d/element-io.list`

Looks like I'm in sync. The trick with "--refresh-keys" from @eighthave comment above worked for me, though: after importing the keys, no complaints from apt.

Thanks.

**Comment by eighthave** [comment]
The setup I'm referring to predates the element keyring package, that's why it 
has a different path.
